authorizations/authorization-jdbc/src/main/java/org/apache/gravitino/authorization/jdbc/JdbcSecurableObjectMappingProvider.java

```diff
@@ -0,0 +1,218 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.gravitino.authorization.jdbc;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.gravitino.MetadataObject;
+import org.apache.gravitino.MetadataObjects;
+import org.apache.gravitino.authorization.AuthorizationMetadataObject;
+import org.apache.gravitino.authorization.AuthorizationPrivilege;
+import org.apache.gravitino.authorization.AuthorizationPrivilegesMappingProvider;
+import org.apache.gravitino.authorization.AuthorizationSecurableObject;
+import org.apache.gravitino.authorization.Privilege;
+import org.apache.gravitino.authorization.SecurableObject;
+
+/**
+ * JdbcSecurableObjectMappingProvider is used for translating securable object to authorization
+ * securable object.
+ */
+public class JdbcSecurableObjectMappingProvider implements AuthorizationPrivilegesMappingProvider {
+
+  private final Map<Privilege.Name, Set<AuthorizationPrivilege>> privilegeMapping =
+      ImmutableMap.of(
+          Privilege.Name.CREATE_TABLE,
+              Sets.newHashSet(JdbcPrivilege.valueOf(JdbcPrivilege.Type.CREATE)),
+          Privilege.Name.CREATE_SCHEMA,
+              Sets.newHashSet(JdbcPrivilege.valueOf(JdbcPrivilege.Type.CREATE)),
+          Privilege.Name.SELECT_TABLE,
+              Sets.newHashSet(JdbcPrivilege.valueOf(JdbcPrivilege.Type.SELECT)),
+          Privilege.Name.MODIFY_TABLE,
+              Sets.newHashSet(
+                  JdbcPrivilege.valueOf(JdbcPrivilege.Type.SELECT),
+                  JdbcPrivilege.valueOf(JdbcPrivilege.Type.UPDATE),
+                  JdbcPrivilege.valueOf(JdbcPrivilege.Type.DELETE),
+                  JdbcPrivilege.valueOf(JdbcPrivilege.Type.INSERT),
+                  JdbcPrivilege.valueOf(JdbcPrivilege.Type.ALTER)),
+          Privilege.Name.USE_SCHEMA,
+              Sets.newHashSet(JdbcPrivilege.valueOf(JdbcPrivilege.Type.USAGE)));
+
+  private final Map<Privilege.Name, PrivilegeScope> privilegeScopeMapping =
+      ImmutableMap.of(
+          Privilege.Name.CREATE_TABLE, PrivilegeScope.TABLE,
+          Privilege.Name.CREATE_SCHEMA, PrivilegeScope.DATABASE,
+          Privilege.Name.SELECT_TABLE, PrivilegeScope.TABLE,
+          Privilege.Name.MODIFY_TABLE, PrivilegeScope.TABLE,
+          Privilege.Name.USE_SCHEMA, PrivilegeScope.DATABASE);
```

Comment:
I think maybe we didn't need import new concept`PrivilegeScope`.
I think we can uses `MetadataObject.Type.SCHEMA` to instead `PrivilegeScope.DATABASE` , and 
uses `MetadataObject.Type.TABLE` to instead `PrivilegeScope.TABLE`.

Reply:
Refactored.